trackers/dataset/utils.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+import supervision as sv
+
+from trackers.log import get_logger  # Added import
+
+logger = get_logger(__name__)  # Added logger instance
+
+
+def relabel_ids(detections: sv.Detections) -> sv.Detections:
+    """
+    Relabels `tracker_id`s to be contiguous integers starting from 0.
+
+    Handles potential NaN or non-integer IDs gracefully. IDs that cannot be
+    processed are left as -1 in the output.
+
+    Args:
+        detections (sv.Detections): The detections object whose `tracker_id`s
+            need relabeling.
+
+    Returns:
+        sv.Detections: The detections object with relabeled `tracker_id`s.
+        Returns the original object if no valid IDs are found or if input is empty.
+    """
+    if len(detections) == 0 or detections.tracker_id is None:
+        return detections
+
+    # 1. Filter out potential NaN values first
```

Comment:
Probably it's best to avoid numbered steps in comments. This is because if the logic gets changed, there's a chance that the numbering would need to change as well.